src/lib/PreviewError.js

```diff
@@ -0,0 +1,29 @@
+class PreviewError extends Error {
+    /** @property {string} - English error code */
+    code;
+
+    /** @property {Object} - Optional error details, will contain the native error under `originalError` if available */
+    details;
+
+    /** @property {string} - Translated error message if error is shown to user */
+    message;
+
+    /**
+     * [constructor]
+     *
+     * @param {string} code - English error code
+     * @param {string} [message] - Error message, should be translated if displayed to the user
+     * @param {Object} [details] - Optional error details
+     * @param {Object} [details.originalError] - If error originates from a native error, this will refer to that error
```

Comment:
I've started a conversation on Slack about copying the native error properties, since `PreviewError` extends `Error`